[WebRTC] Split most generic parts of LibWebRTCNetwork into a reusable WebRTCNetwork super class

philn opened the pull request:
#### 4bdca5ffb3004a9b79b0b7b383417b8e6d21021d
<pre>
[WebRTC] Split most generic parts of LibWebRTCNetwork into a reusable WebRTCNetwork super class
<a href="https://bugs.webkit.org/show_bug.cgi?id=299094">https://bugs.webkit.org/show_bug.cgi?id=299094</a>

Reviewed by Youenn Fablet.

Refactor the libwebrtc-agnostic code from LibWebRTCNetwork into a WebRTCNetworkBase
class (WebRTCNetwork is used for a namespace already) and remove USE(LIBWEBRTC) ifdefs from
WebMDNSRegister, making it usable later on for the WPE and GTK ports when GstWebRTC is enabled.

Covered by existing tests.

* Source/WebKit/Sources.txt:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCNetwork.cpp:
(WebKit::LibWebRTCNetwork::LibWebRTCNetwork):
(WebKit::LibWebRTCNetwork::setAsActive):
(WebKit::LibWebRTCNetwork::setConnection):
(WebKit::LibWebRTCNetwork::dispatch):
(WebKit::LibWebRTCNetwork::ref const): Deleted.
(WebKit::LibWebRTCNetwork::deref const): Deleted.
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCNetwork.h:
* Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.cpp:
(WebKit::WebMDNSRegister::WebMDNSRegister):
(WebKit::WebMDNSRegister::ref const):
(WebKit::WebMDNSRegister::deref const):
* Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.h:
* Source/WebKit/WebProcess/Network/webrtc/WebRTCNetworkBase.cpp: Added.
(WebKit::WebRTCNetworkBase::WebRTCNetworkBase):
(WebKit::WebRTCNetworkBase::~WebRTCNetworkBase):
(WebKit::WebRTCNetworkBase::ref const):
(WebKit::WebRTCNetworkBase::deref const):
(WebKit::WebRTCNetworkBase::setAsActive):
* Source/WebKit/WebProcess/Network/webrtc/WebRTCNetworkBase.h: Copied from Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.h.
(WebKit::WebRTCNetworkBase::networkProcessCrashed):
(WebKit::WebRTCNetworkBase::isActive const):
(WebKit::WebRTCNetworkBase::mdnsRegister):
(WebKit::WebRTCNetworkBase::protectedMDNSRegister):

Canonical link: <a href="https://commits.webkit.org/300740@main">https://commits.webkit.org/300740@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3e896f201c6ab506e127aab93c8a1d77a9ae2a5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121769 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41470 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32134 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128303 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73859 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50061 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92531 "Failed to checkout and rebase branch from PR 50994") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61509 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124721 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33647 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109056 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73190 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32659 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71816 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103147 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27404 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131084 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48704 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37037 "Found 1 new test failure: compositing/visible-rect/flipped-preserve-3d.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101116 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49076 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105273 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100988 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26029 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46353 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24465 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45405 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48562 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54289 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48032 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51381 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49714 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9701d6d5fa653cc124dca45fcef3d57b0216eb32. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50994

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3a648ba644350dab2ba928e765a9811b100e2956. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50994

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b3e896f201c6ab506e127aab93c8a1d77a9ae2a5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50994

philn committed:
[WebRTC] Split most generic parts of LibWebRTCNetwork into a reusable WebRTCNetwork super class
https://bugs.webkit.org/show_bug.cgi?id=299094

Reviewed by Youenn Fablet.

Refactor the libwebrtc-agnostic code from LibWebRTCNetwork into a WebRTCNetworkBase
class (WebRTCNetwork is used for a namespace already) and remove USE(LIBWEBRTC) ifdefs from
WebMDNSRegister, making it usable later on for the WPE and GTK ports when GstWebRTC is enabled.

Covered by existing tests.

* Source/WebKit/Sources.txt:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCNetwork.cpp:
(WebKit::LibWebRTCNetwork::LibWebRTCNetwork):
(WebKit::LibWebRTCNetwork::setAsActive):
(WebKit::LibWebRTCNetwork::setConnection):
(WebKit::LibWebRTCNetwork::dispatch):
(WebKit::LibWebRTCNetwork::ref const): Deleted.
(WebKit::LibWebRTCNetwork::deref const): Deleted.
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCNetwork.h:
* Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.cpp:
(WebKit::WebMDNSRegister::WebMDNSRegister):
(WebKit::WebMDNSRegister::ref const):
(WebKit::WebMDNSRegister::deref const):
* Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.h:
* Source/WebKit/WebProcess/Network/webrtc/WebRTCNetworkBase.cpp: Added.
(WebKit::WebRTCNetworkBase::WebRTCNetworkBase):
(WebKit::WebRTCNetworkBase::~WebRTCNetworkBase):
(WebKit::WebRTCNetworkBase::ref const):
(WebKit::WebRTCNetworkBase::deref const):
(WebKit::WebRTCNetworkBase::setAsActive):
* Source/WebKit/WebProcess/Network/webrtc/WebRTCNetworkBase.h: Copied from Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.h.
(WebKit::WebRTCNetworkBase::networkProcessCrashed):
(WebKit::WebRTCNetworkBase::isActive const):
(WebKit::WebRTCNetworkBase::mdnsRegister):
(WebKit::WebRTCNetworkBase::protectedMDNSRegister):

Canonical link: https://commits.webkit.org/300740@main
